crates/matrix-sdk/src/sliding_sync/cache.rs

```diff
@@ -174,9 +174,8 @@ pub(super) async fn restore_sliding_sync_state(
 
             // Let's update the `SlidingSync`.
             if let Some(since) = to_device_since {
-                if let Some(to_device_ext) =
-                    extensions.get_or_insert_with(Default::default).to_device.as_mut()
-                {
+                let to_device_ext = &mut extensions.get_or_insert_with(Default::default).to_device;
+                if to_device_ext.enabled == Some(true) {
```

Comment:
I'm just not sure of the code here. I don't understand when the since is supposed to be added.

Reply:
This is an interesting one. I think using `ExtensionConfig` to track whether or not a certain extension is supposed to be enabled might actually be a bad idea..

I'm pretty sure that this will work in practice right now, since we always set `Some(true)` for the relevant extensions when they're being enabled though. If we do the other change I mentioned, we could change this to `!= Some(false)` presumably.